src/js/core/widget/core/Spin.js

```diff
@@ -193,14 +193,20 @@
 			var self = this,
 				direction = delta > 0 ? 1 : -1,
 				borderItem,
-				newItemToPlace;
+				newItemToPlace,
+				halfOfFreeCarouselItems = Math.round((self._numberOfCarouselItems - self._ui.items.length) / 2);
```

Comment:
did the issue continue even before halfOfFreeCarouselItems was applied?
or did it happen intermittently?

Reply:
The problem was always when number of spin elements was < NUMBER_OF_CAROUSEL_ITEMS. However, initially the NUMBER_OF_CAROUSEL_ITEMS was equal 5 or 7 while the number picker had 10+ values and the issue was not so visible.